homeassistant/components/modbus/const.py

```diff
@@ -12,6 +12,7 @@
 CONF_CURRENT_TEMP_REGISTER_TYPE = "current_temp_register_type"
 CONF_DATA_COUNT = "data_count"
 CONF_DATA_TYPE = "data_type"
+CONF_DATA_SIZE = "data_size"
```

Comment:
I think you mean “register size”, because data size would be the size of the complete structure.

Reply:
Expected data size

Reply:
No what happens if you configure datatype: int32, that has fixed size. what you want in your case is more like
```
datatype: int32
count: 1
register_size: 4
```

Remember I can define a struct that contains e.g 4 int32, that would be:
```
data_type: int32
count: 4
register_size: 4
```
data_size would be 16 and then you would expect 16bytes in each register not 4bytes in 4 registers.